### ISSUE TITLE
Ecto Sandbox, Projections and In Memory adapter

### ISSUE THREAD

**Issue by roehst** [commanded/commanded#553]
Hi all,

We are trying to read projections during test.

We have just created our first projector, but we are having trouble reading projections during the test.

In production we use Postgres as event store, and in test we use the in memory adapter.

The testing guide (https://github.com/commanded/commanded/wiki/Testing-your-application) in the Wiki provides instructions for how to keep projections up-to-date running tests, involving issuing a TRUNCATE command to event store. 

Our main pattern to run commands is to dispatch a command with strong consistency, and then read it back from the projection before returning.

With the projection, it returns nil (i.e., it doesn't find the rows in the projections).

So I assume there might be some step akin to truncating tables in the Postgres adapter.

I tried reading the source code for the in memory manager, and I can see it dispatches events for subscribers. I am assuming this can have something to do as well.

So, am I missing a step here? If my writing is not sufficiently clear, don't feel afraid to ask for more clarity. Thanks in advance!

**Comment by roehst** [comment]
Hey @drteeth (cool name btw), your explanation makes sense, and thanks for it.

We tried the allow configuration as per Ecto's documentation, and we are uncertain as to proceed.

We tried that, but we were still getting no projections from the projector after we run the command (with strong consistency).

What we did was, in the test file, set the mode to shared, with the current process as pid and the pid of the projector as allow. We found the projectors pid via which_children on its supervisor.

Maybe we should start the projector manually at the beginning of the test? Do you have more ideas we can try?

**Comment by drteeth** [comment]
Yes starting the projector at the start of your test makes sense. use:
```elixir
pid = start_supervised(YourProjector)
```

so that it is properly cleaned up by the test
